.github/workflows/test-pull-push-platforms.yml

```diff
@@ -0,0 +1,74 @@
+name: Test Pull/Push Platforms
+on:
+  pull_request:
+    paths:
+      - "go.*"
+      - "pkg/**"
+      - "cmd/**"
+      - "Makefile"
+      - "vendor/**"
+      - ".github/workflows/**"
+      - "!.github/workflows/docs**"
+  push:
+    branches: [ main, master ]
+  workflow_dispatch:
+    inputs:
+      debug_enabled:
+        description: 'Run the build with tmate set "debug_enabled"'
+        type: boolean
+        required: false
+        default: false
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+permissions:
+  contents: read
+
+jobs:
+  prepare-testargs:
+    runs-on: ubuntu-latest
+    outputs:
+      testargs: ${{ steps.compute-testargs.outputs.testargs }}
+    steps:
+      - name: Compute TESTARGS
+        id: compute-testargs
+        run: |
+          # Run all tests that limit their scope when running with any value of GOTEST_SHORT
+          TP=(
+            "TestAppdirAlreadyInUse"
+            "TestAutocomplet"
+            "TestAvailableUpdates"
+            "TestComposerCreateCmd"
+            "TestCmdDescribe"
+            "TestCmdList"
+            "TestCmdStart"
+            "TestCmdStop"
+            "TestConfigDatabaseVersion"
+            "TestDdevAllDatabases"
+            "TestDdevFullSite.*"
+            "TestDdevImportFiles.*"
+            "TestDdevUploadDir.*"
+            "TestDdevXdebugEnabled"
+            "TestDdevXhprofPrependEnabled"
+            "TestDdevXhprofXhguiEnabled"
+            "TestGetApps.*"
+            "TestPHPConfig"
+            "TestPHPWebserverType"
+            "TestWriteDrushConfig"
+            "Test.*(Push|Pull)"
+          )
+          TESTARGS="-failfast -run '($(IFS="|"; echo "${TP[*]}"))'"
+          echo "pull-push-test-platforms TESTARGS=$TESTARGS"
+          echo "testargs=${TESTARGS}" >> $GITHUB_ENV
+
+  pull-push-test-platforms:
```

Comment:
The reality is it would be better to do the push/pull and test-everything-with-all-CMSs as separate workflows. But we don't have to do that now.

Reply:
No, I'm talking about what push-pull already does, running some tests (TestDdevFullSiteSetup and TestDdevAllDatabases) with all permutations. I'm saying that that would be better as a separate workflow, separated from the push-pull tests.